engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/decimal/RoundNode.java

```diff
@@ -0,0 +1,103 @@
+package org.enso.interpreter.node.expression.builtin.number.decimal;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.profiles.CountingConditionProfile;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.node.expression.builtin.number.utils.BigIntegerOps;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.builtin.Builtins;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+
+@BuiltinMethod(
+        type = "Decimal",
+        name = "round",
+        description = "Decimal ceiling, converting to a small or big integer depending on size.")
+public class RoundNode extends Node {
+    private final CountingConditionProfile fitsProfile = CountingConditionProfile.create();
+
+    /** Minimum value for the `decimal_places` parameter to `roundDouble`. */
+    private static final double ROUND_MIN_DECIMAL_PLACES = -15;
+
+    /** Maximum value for the `decimal_places` parameter to `roundDouble`. */
+    private static final double ROUND_MAX_DECIMAL_PLACES = 15;
+
+    /** Minimum value for the `n` parameter to `roundDouble`. */
+    private static final double ROUND_MIN_DOUBLE = -99999999999999.0;
+
+    /** Minimum value for the `n` parameter to `roundDouble`. */
+    private static final double ROUND_MAX_DOUBLE = 99999999999999.0;
+
+    Object execute(double n, long decimalPlaces, boolean useBankers) {
+        if (decimalPlaces < ROUND_MIN_DECIMAL_PLACES || decimalPlaces > ROUND_MAX_DECIMAL_PLACES) {
+            decimalPlacesOutOfRangePanic(decimalPlaces);
```

Comment:
I'd change this to `throw decimalPlacesOutOfRangePanic....` to tell the compiler that the execution of this method terminates after this `if` branch is taken.